### PR TITLE
histogramdd normed argument replaced by density

### LINKED ISSUE #3975: [upstream change] histogramdd normed has been removed
normed has been removed in favour of density in numpy 1.24. Which is making CI fail.

Strangely enough this was done 7 months ago and we didn't pick this up at all in our Cron jobs using nightly builds until yesterday, so we need to look at why this is happening (it's possible the install order is wrong).

### PR DESCRIPTION
**Pull request by IAlibay**
Fixes #3975 

Changes made in this Pull Request:
 - 

PR Checklist
------------
 - [x] Tests?
 - [x] Docs?
 - [x] CHANGELOG updated?
 - [x] Issue raised/referenced?

### CODE REVIEW

**Comment by pgbarletta** [comment]
I think you're just missing this [normed](https://github.com/MDAnalysis/mdanalysis/blob/123dabbe10764bfd6431aea445c3e96069c10041/package/MDAnalysis/analysis/density.py#L476). And [this one](https://github.com/MDAnalysis/mdanalysis/blob/123dabbe10764bfd6431aea445c3e96069c10041/package/MDAnalysis/analysis/density.py#L647) on the docs.

File: package/CHANGELOG [hunk 1]
```diff
@@ -18,6 +18,8 @@ The rules for this file:
  * 2.5.0
 
 Fixes
+  * np.histogramdd calls in :class:`DensityAnalysis` now pass the `density`
```
Comment (IAlibay) [inline package/CHANGELOG]: @MDAnalysis/coredevs please look at discord for this one - in my opinion we probably should do a 2.4.2 release since this has effectively killed the DensityAnalysis code. However, _how_ we do bugfix releases really should be discussed.

TLDR; I'm pushing for us to be able to do bugfixes from a separate feature branch than `develop`. That way PRs can keep rolling into `develop` without breaking semver if we need to do a bugfix release.

**Comment by IAlibay** [comment]
So the consensus (from @orbeckst and @richardjgowers that spoke up on this), is that we'll do separate bugfix branches based off the `release-2.4.x` tags where we cherry-pick commits from develop (where possible) and do release from them (please correct me if I misinterpreted things!).

As such, I've moved the changelog entry here to be 2.4.2, with the aim that I'll just cherry-pick the squash-merged commit out of here and re-deploy (yay..).

@MDAnalysis/coredevs please review this as a priority - it's the main cause of current CI failures, and it's blocking the 2.4.x announcement (we can't announce it if it's failing as soon as it's released).

**Review by orbeckst — APPROVED** [review]
thank you, lgtm

**Review by hmacdope — APPROVED** [review]
LGTM

**Comment by hmacdope** [comment]
@IAlibay only concern is the 32 bit build? Causing some failures on the Azure CI

File: package/CHANGELOG [hunk 2]
```diff
@@ -26,6 +26,15 @@ Changes
 Deprecations
 
 
+12/21/22 IAlibay
+
+ * 2.4.2
+
+Fixes
+  * np.histogramdd calls in :class:`DensityAnalysis` now pass the `density`
+    argument rather than the NumPy 1.24 removed `density` (PR #3976)
```
Comment (tylerjereddy) [inline package/CHANGELOG]: you may have meant `normed` here at the end?

**Review by tylerjereddy — COMMENTED** [review]
CI failure looks like `h5py` not supporting NumPy `1.24.0` just yet on that platform.

32-bit Windows binaries seem to be getting dropped upstream a lot lately.

Releasing off of maintenance branches is what I do for SciPy and what Chuck does for NumPy, so +1 for that.

**Comment by IAlibay** [comment]
we've had this particular h5py issue for some while (pre numpy 1.24 release), but the fact that there are more now might indicate that it's either related or it has accentuated the problem. I'm going to try switching the pipeline up a little bit to limit things for 32bit.

**Comment by IAlibay** [comment]
Only extra thing I've done is slightly change the azure pipelines to pin numpy for 32bit (hopefully this fixes it 🤞🏽 thanks @tylerjereddy !), so with two approvals I'm going to go ahead with the merge.
